src/tests/nativeaot/SmokeTests/UnitTests/Generics.cs

```diff
@@ -2434,6 +2435,97 @@ public static void Run()
         }
     }
 
+    class TestFunctionPointerLoading
```

Comment:
We can use this test coverage on runtime flavors. Can this be regular tests that runs on all runtimes instead? (Either under src\tests or under src\libraries is fine.)

Reply:
I'm worried about people "cleaning up" the `MakeGeneric` part or the statics. This test is written in a very specific way because it's for Native AOT. Libraries devs don't know about these subtleties.

Reply:
(To be clear the concern is that this will be "cleaned up" in one of the 3000-line diffs when some analyzer enthusiast enables a new analyzer. I do look at things related to NativeAOT but I don't have the bandwidth to track the whole repo. The analyzer changes tend to cause subtle damage like this.)